src/components/application_manager/src/commands/command_impl.cc

```diff
@@ -123,7 +125,9 @@ bool CommandImpl::ReplaceMobileWithHMIAppId(
         std::set<std::string>::const_iterator key = keys.begin();
         for (; key != keys.end(); ++key) {
           std::string k = *key;
-          ReplaceMobileWithHMIAppId(message[*key]);
+          if (!ReplaceMobileWithHMIAppId(message[*key])) {
+            break;
```

Comment:
break  cause  the function return true, however, it shall return false.

Reply:
@yang1070 
Done with 866dfe8

Reply:
@yang1070 
Reverted previous changes due to issues with ATF tests. As I found there is a mixed logic for two use cases: first is related to bugfix done for Ford Sync transport switch implementation which requires checking of application with app_id, second is common usage when app_id is replaced during message processing. So I propose to leave code as is for now and create another issue for investigation how this part can be re-factored.